test/e2e/certs_test.go

```diff
@@ -21,24 +25,52 @@ func TestRouterCerts(t *testing.T) {
 	require.NoError(t, err)
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	require.NoError(t, err)
-	configClient, err := versioned.NewForConfig(kubeConfig)
+	configClient, err := configclient.NewForConfig(kubeConfig)
 	require.NoError(t, err)
-	// machine managed router-certs secret should exist
-	machineManagedRouterCerts, err := kubeClient.CoreV1().Secrets("openshift-config-managed").Get("router-certs", metav1.GetOptions{})
+	operatorClientset, err := operatorclient.NewForConfig(kubeConfig)
 	require.NoError(t, err)
-	// operator managed router-certs secret should exist
-	operatorManagedRouterCerts, err := kubeClient.CoreV1().Secrets("openshift-authentication").Get("v4-0-config-system-router-certs", metav1.GetOptions{})
+	// make sure cluster operator is settled before continuing
+	err = e2e.WaitForClusterOperatorAvailableNotProgressingNotDegraded(t, configClient.ConfigV1(), "authentication")
 	require.NoError(t, err)
-	// operator managed router-certs data should be a copy of machine managed router-certs
-	require.Equal(t, operatorManagedRouterCerts.Data, machineManagedRouterCerts.Data)
-	// there should be an entry for the ingress domain
-	ingress, err := configClient.ConfigV1().Ingresses().Get("cluster", metav1.GetOptions{})
+	routerCertsSecret, err := kubeClient.CoreV1().Secrets("openshift-config-managed").Get("router-certs", metav1.GetOptions{})
 	require.NoError(t, err)
-	require.Contains(t, operatorManagedRouterCerts.Data, ingress.Spec.Domain)
-	// the entry should contain certificates that validate the oauth service hostname
-	block, _ := pem.Decode(operatorManagedRouterCerts.Data[ingress.Spec.Domain])
-	cert, err := x509.ParseCertificate(block.Bytes)
+	// put bad certs in data
```

Comment:
This seems to be racing against the ingress operator which will fix the cert at some point.  i.e. this test may be flakey.